File: public/examples/integrations/toolkits/box_api/retrieve_shared_web_link_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "BoxApi.RetrieveSharedWebLink"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=TOOL_NAME
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'shared_link_header': 'shared_link=https://example.com/s/abc123&shared_link_password=passw0rd',
    'include_attributes_in_response': ['id', 'name', 'url', 'modified_at'],
    'etag_if_updated_only': '"d41d8cd98f00b204e9800998ecf8427e"'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/toolkits/box_api/retrieve_shared_web_link_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "BoxApi.RetrieveSharedWebLink";

// Start the authorization process
const authResponse = await client.tools.authorize({
    tool_name: TOOL_NAME,
    user_id: USER_ID
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "shared_link_header": "shared_link=https://example.com/s/abc123&shared_link_password=passw0rd",
  "include_attributes_in_response": [
    "id",
    "name",
    "url",
    "modified_at"
  ],
  "etag_if_updated_only": "\"d41d8cd98f00b204e9800998ecf8427e\""
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
